report_github_metrics.rb

```diff
@@ -40,8 +40,15 @@ def submit_metrics(metrics, datadog_client, metric_prefix)
   end
 end
 
-def prior_jobs(jobs)
-  jobs.select{|job| !job["conclusion"].nil? }
+def prior_jobs(github_client, jobs)
+  length = jobs[:total_count].to_i
+  finished_jobs = jobs[:jobs].select{|job| !job["conclusion"].nil? }
+  while length - jobs[:jobs].length > 0
+    length -= jobs[:jobs].length
+    jobs = github_client.get(github_client.last_response.rels[:next].href)
+    finished_jobs += jobs[:jobs].select{|job| !job["conclusion"].nil? }
+  end
+  finished_jobs
```

Comment:
Might be useful to also print a warning if `finished_jobs.length != jobs[:total_count].to_i`.

Reply:
I think we wouldn't actually expect this to be true because, there could be unfinished jobs that are filtered out and not important. For example the action filters the job it is running in out as non complete.

Reply:
But besides the job the action is running in, don't we usually want to capture stats for all other jobs? i.e. if there is another unfinished job at the time of the action, we probably should update the action `needs` to run after that. If so, maybe the warning should be based on `finished_jobs.length + 1 != jobs[:total_count].to_i`. Just want to make sure we wont miss jobs quietly.

Reply:
Thanks, that makes sense 👍